protocols/kad/src/handler.rs

```diff
@@ -766,15 +777,31 @@ where
             ));
         }
 
-        if let Poll::Ready(Some(event)) = self.outbound_substreams.poll_next_unpin(cx) {
+        let num_in_progress_outbound_substreams =
+            self.active_outbound_substreams.len() + self.num_upgrading_outbound_streams;
+        for _ in 0..MAX_NUM_SUBSTREAMS.saturating_sub(num_in_progress_outbound_substreams) {
+            if let Some(outbound_substream) = self.next_outbound_streams.pop_front() {
+                self.active_outbound_substreams.push(outbound_substream);
```

Comment:
The `outbound_substream` being pushed into this channel directly further below yields the `OutboundSubstreamRequest` and then moves to `OutboundSubstreamState::Done`. Does it need to be added to `active_outbound_substreams` at all then? Why not increase `num_upgrading_outbound_streams` right away and emit the `OutboundSubstreamRequest` directly?